### ISSUE TITLE
Update to v1.4.0 fails.

### ISSUE THREAD

**Issue by fotis400** [thingsboard/thingsboard#629]
Hi, I just tried to update my Thingsboard installation, but it failed.

I had a properly working installation of v1.3.1, and tried to update it to v1.4.0. I followed closely the instructions [here](https://thingsboard.io/docs/user-guide/install/upgrade-instructions/#upgrading-to-140). Everything went ok (apart from the fact that the package manager never prompted me to merge anything), till I reached the step with the upgrade script.

I execute:
`$ sudo /usr/share/thingsboard/bin/install/upgrade.sh --fromVersion=1.3.1`

But the output is:
```
 ===================================================
 :: ThingsBoard ::       (v1.4.0)
 ===================================================

Error creating bean with name 'thingsboardInstallService': Unsatisfied dependency expressed through field 'databaseUpgradeService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'sqlDatabaseUpgradeService': Unsatisfied dependency expressed through field 'dashboardService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'dashboardServiceImpl': Unsatisfied dependency expressed through field 'relationService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'baseRelationService': Unsatisfied dependency expressed through field 'entityService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'baseEntityService': Unsatisfied dependency expressed through field 'deviceService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'deviceServiceImpl': Unsatisfied dependency expressed through field 'cacheManager'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cacheManager' defined in class path resource [org/thingsboard/server/dao/cache/CaffeineCacheConfiguration.class]: Invocation of init method failed; nested exception is java.lang.NullPointerException
ThingsBoard upgrade failed!
```

The server is not working any more.

[Here](https://pastebin.com/gwQvpkAU) is the application log.

[Here](https://pastebin.com/qGYQ4LEH) is the installation log.

Any ideas on fixing this?

**Comment by fotis400** [comment]
Forgot to mention that I am using a Postgresql database.

**Comment by vparomskiy** [comment]
Please verify your thingsboard.yml configuration before running upgrade.sh
It should contain caching configuration
```
cache:
  # caffeine or redis
  type: "${CACHE_TYPE:caffeine}"

caffeine:
  specs:
    relations:
      timeToLiveInMinutes: 1440
      maxSize: 100000
    deviceCredentials:
      timeToLiveInMinutes: 1440
      maxSize: 100000
    devices:
      timeToLiveInMinutes: 1440
      maxSize: 100000
```
The problem is that system could not properly start caching service.
I think the reason is hidden in in this

> apart from the fact that the package manager never prompted me to merge anything

Required config was not merged into your existing config during package upgrade. So you need to manually check that everything is up to date. And only after that run this command

`sudo /usr/share/thingsboard/bin/install/upgrade.sh --fromVersion=1.3.1`

**Comment by fotis400** [comment]
I did the fix in the config, but the update still fails.

This "merge" is only about the config file? Or should I check other things too?

Does the following have to remain in the config file, or should I only keep the lines you provided me?
```
  # Enable/disable cache functionality.
  enabled: "${CACHE_ENABLED:false}"
  device_credentials:
    # Default time to store device credentials in cache, in seconds
    time_to_live: "${CACHE_DEVICE_CREDENTIAL_TTL:3600}"
    # Maximum size of the map. When maximum size is reached, the map is evicted based on the policy defined.
    max_size:
      # Max size policy options:
      # PER_NODE: Maximum number of map entries in each JVM.
      # PER_PARTITION: Maximum number of map entries within each partition.
      # USED_HEAP_SIZE: Maximum used heap size in megabytes for each JVM.
      # USED_HEAP_PERCENTAGE: Maximum used heap size percentage for each JVM.
      # FREE_HEAP_SIZE: Minimum free heap size in megabytes for each JVM.
      # FREE_HEAP_PERCENTAGE: Minimum free heap size percentage for each JVM.
      policy: "${CACHE_DEVICE_CREDENTIAL_MAX_SIZE_POLICY:PER_NODE}"
      size: "${CACHE_DEVICE_CREDENTIAL_MAX_SIZE_SIZE:1000000}"
```

The new error message is:
```
 ===================================================
 :: ThingsBoard ::       (v1.4.0)
 ===================================================

Error creating bean with name 'jpaTimeseriesDao': Injection of autowired dependencies failed; nested exception is java.lang.IllegalArgumentException: Could not resolve placeholder 'sql.ts_inserts_executor_type' in string value "${sql.ts_inserts_executor_type}"

```

**Comment by vparomskiy** [comment]
Here is the latest version
https://github.com/thingsboard/thingsboard/blob/release-1.4/application/src/main/resources/thingsboard.yml

In short, those changes were made:
1. add mqtt & coap enable flag
2. add quota configuration
3. add sql configuration parameters
4. change cache configuration (caffeine/redis)
5. add audit_log configuration

You will need to apply all those changes

**Comment by fotis400** [comment]
Ok that did the trick. I manually merged the new file with my own.

One question. I noticed that there is a new file generated, named `thingsboard.yml.rpmnew`. Is this nfile needed, or it is safe to delete it?

**Comment by vparomskiy** [comment]
You can remove it

**Comment by fotis400** [comment]
Thank you for your help!
